action.php

```diff
@@ -68,27 +67,36 @@ protected function html()
         global $INPUT;
         global $USERINFO;
         $id = $INPUT->str('id');
+        $ackSubmitted = $INPUT->str('ack') === 'true';
         $user = $INPUT->server->str('REMOTE_USER');
         if ($id === '' || $user === '') return '';
 
         /** @var helper_plugin_acknowledge $helper */
         $helper = plugin_load('helper', 'acknowledge');
 
+        if ($ackSubmitted) {
+            $helper->saveAcknowledgement($id, $user);
+        }
+
         $html = '';
 
         $ack = $helper->hasUserAcknowledged($id, $user);
         if ($ack) {
 
             $html .= '<div>';
-            $html .= 'You acknowledged this page ' . sprintf('%f', $ack);
+            $html .= $this->getLang('ackGranted') . sprintf('%s', dformat($ack));
             $html .= '</div>';
         } elseif ($helper->isUserAssigned($id, $user, $USERINFO['grps'])) {
+            $form = new Form(['id' => 'ackForm']);
+            $form->addCheckbox('ack');
```

Comment:
The checkbox should get the `required` attribute

Reply:
right